pragma-node/src/utils/signing/starkex.rs

```diff
@@ -0,0 +1,178 @@
+use bigdecimal::{BigDecimal, ToPrimitive};
+use starknet::core::{
+    crypto::pedersen_hash, types::FieldElement, utils::cairo_short_string_to_felt,
+};
+
+use pragma_common::types::ConversionError;
+
+use super::Signable;
+
+pub struct StarkexPrice {
+    pub oracle_name: String,
+    pub pair_id: String,
+    pub timestamp: u64,
+    pub price: BigDecimal,
+}
+
+impl StarkexPrice {
+    pub fn get_global_asset_id(pair_id: &str) -> Result<String, ConversionError> {
```

Comment:
why don't we use `self.pair_id` here ?

Reply:
I'm also using the function elsewhere when encoding the asset id:
```rust
impl TryFrom<MedianEntryWithComponents> for AssetOraclePrice {
    type Error = ConversionError;

    fn try_from(median_entry: MedianEntryWithComponents) -> Result<Self, Self::Error> {
        let signed_prices: Result<Vec<SignedPublisherPrice>, ConversionError> = median_entry
            .components
            .into_iter()
            .map(SignedPublisherPrice::try_from)
            .collect();

        let global_asset_id = StarkexPrice::get_global_asset_id(&median_entry.pair_id)?;
```